### PR TITLE
Add option to use hf tokenizer

### PR DESCRIPTION
**Pull request by RissyRan**
# Description

Add option to use hf tokenizer

# Test

* provide invalid hf tokenizer, it will failed to find right tokenizer

```
python JetStream/benchmarks/benchmark_serving.py   \
--tokenizer ~/maxtext/assets/tokenizer.mistral-v3  \
--save-result   \
--save-request-outputs   \
--request-outputs-file-path outputs-scanned-8x22b-hf-4.json   \
--num-prompts 4   \
--max-output-length 1024   \
--dataset openorca --run-eval true --use-hf-tokenizer=True


Using HuggingFace tokenizer: /home/ranran/maxtext/assets/tokenizer.mistral-v3
Traceback (most recent call last):
  File "/home/ranran/venv-maxtext/lib/python3.10/site-packages/transformers/utils/hub.py", line 403, in cached_file
    resolved_file = hf_hub_download(
  File "/home/ranran/venv-maxtext/lib/python3.10/site-packages/huggingface_hub/utils/_validators.py", line 106, in _inner_fn
    validate_repo_id(arg_value)
  File "/home/ranran/venv-maxtext/lib/python3.10/site-packages/huggingface_hub/utils/_validators.py", line 154, in validate_repo_id
    raise HFValidationError(
huggingface_hub.errors.HFValidationError: Repo id must be in the form 'repo_name' or 'namespace/repo_name': '/home/ranran/maxtext/assets/tokenizer.mistral-v3'. Use `repo_type` argument if needed.
```

* provide valid hf tokenizer, it will pass

```
python JetStream/benchmarks/benchmark_serving.py   \
--tokenizer mistralai/Mixtral-8x22B-Instruct-v0.1  \
--save-result   \
--save-request-outputs   \
--request-outputs-file-path outputs-scanned-8x22b-hf-4.json   \
--num-prompts 4   \
--max-output-length 1024   \
--dataset openorca --run-eval true --use-hf-tokenizer=True

Using HuggingFace tokenizer: mistralai/Mixtral-8x22B-Instruct-v0.1
len(sampled_indices)=4
In InputRequest, pass in actual output_length for each sample
The dataset contains 4 samples.
The filtered dataset contains 4 samples.
  0%|                                                                                                                               | 0/4 [00:00<?, ?it/s]

```

### CODE REVIEW

File: benchmarks/benchmark_serving.py [hunk 1]
```diff
@@ -156,16 +157,26 @@ def to_dict(self):
...
+    tokenizer_name: str,
+    use_hf_tokenizer: bool,
+) -> Any:
   """Return a tokenizer or a tokenizer placholder."""
   if tokenizer_name == "test":
+    print("Using test tokenizer")
     return "test"
+  elif use_hf_tokenizer:
+    print(f"Using HuggingFace tokenizer: {tokenizer_name}")
+    return AutoTokenizer.from_pretrained(tokenizer_name)
```
Comment (vipannalla) [inline benchmarks/benchmark_serving.py:171]: User needs to accept the MixtralAI (or other model) agreeement on HuggingFace, else this line will error out. Can you add a comment about it for future users?

**Review by vipannalla — APPROVED** [review]
LGTM, thanks

File: benchmarks/benchmark_serving.py (hunk 1 above)
Comment (RissyRan) [reply benchmarks/benchmark_serving.py:171]: Sure thing!

Comment (RissyRan) [reply benchmarks/benchmark_serving.py:174]: Added comment accordingly.

**Pull request merged**